### ISSUE TITLE
Projects need to be able to specify manifest location - determining it by "navigating up" is problematic

### ISSUE THREAD

**Issue by aggieNick02** [microsoft/vcpkg#20690]
**Is your feature request related to a problem? Please describe.**
Currently the manifest used for a project is determined by looking next to the project, and then navigating up directories, until vcpkg.json is found. In a standard flat solution which has many projects all at the same level, this is really inflexible. It requires either manifest and installed package duplication, or reorganizing solution file structure. It is not unreasonable or uncommon to have a solution with several related projects that need to use one of two triplets (e.g., some projects use x64-windows, some use x64-windows-static). There is not a good option available for using manifests in such solutions.

**Proposed solution**
Allow projects to specify the directory that contains their vcpkg.json file

**Describe alternatives you've considered**
As mentioned in the problem statement, you can put individual per-project vcpkg.json files next to each project, but this means any dependency changes must be copy pasted, and duplicates the installed ports for each project

**Additional context**
By combining this feature with allowing overlay port and overlay triplet directories to be specified within a vcpkg.json (https://github.com/microsoft/vcpkg/issues/12289), manifests would be a viable option for many more clients. Instead, clients have to come up with their own solutions, and the recent versioning capability is not available to them.

**Comment by aggieNick02** [comment]
Just saw the CLI has --x-manifest-root - maybe this is all I need. I'll play with it and report back.

**Comment by aggieNick02** [comment]
Alright, so `--x-manifest-root` isn't enough because it doesn't tell visual studio where to look for the manifest - it only tells vcpkg. But I can set `VcpkgManifestRoot` in the `vcxproj` and it works pretty well. It gets me halfway to how I'd like things to work. 

(In my setup, `vcpkg` is a subrepo in my repo, so I want all of my projects in that repo to point to use the `vcpkg.exe` in the folder that sits next to them, and to use one of the manifests that are present at the top-level in the repo too. With `VcpkgManifestRoot`, I can get it to use those manifests just fine.)

The remaining problem is getting my projects to use the local vcpkg folder. This means I need to use per-project integration instead of site-wide. Using the nuget mechanism for this works fine, but isn't something I want to check into source code control because the name of the nuget package is based on its local path on disk. The vcpkg FAQ (https://vcpkg.readthedocs.io/en/latest/about/faq/) says I can accomplish the same thing by using the lower level mechanism of just importing the `vcpkg.targets` file in the `vcxproj`. I tried this, and  the project acts like it has vcpkg integration, but it ignores values I set for various vcpkg variables - namely, `VcpkgManifestRoot` is no longer honored. It always looks in the project directory and up regardless of the variable's value.

Is importing the `vcpkg.targets` to get per-project integration still supported or perhaps somehow broken right now? Here's a screenshot of `VcpkgManifestRoot` not being honored when I do it:

![image](https://user-images.githubusercontent.com/16806169/137251512-d76164d6-4f29-47ab-9750-120628234b4d.png)

**Comment by dg0yt** [comment]
> Most projects are typical using the dynamic runtime and all kinds of dependencies (e.g., triplet `x64-windows`), but I have a couple utilities that need the c++ runtime linked statically (triplet `x64-windows-static`). The set of dependencies for the typical projects is pretty large (lots of boost, ffmpeg, etc). The set of dependencies for the little utilities using static runtime is small.

Did you consider using a single custom triplet based on `x64-windows`, which sets static linkage for the desired set of little utilities?

**Comment by aggieNick02** [comment]
Do you mean to do that and then use that triplet for all of my projects to avoid having to build all of the dependencies with the static runtime? The dependencies of the little utilities intersect with the dependencies of the typical projects, so there are some dependencies that need to be built both with static and dynamic runtime linkage, which isn't workable within a single triplet.

Apologies if I'm misunderstanding the suggestion and you meant something different.

**Comment by aggieNick02** [comment]
I've played enough that with the various things pointed out by @autoantwort in addition to setting `$(VcpkgManifestRoot)` manually in the vcxproj file allows me to do what I wanted. I have everything working now except for a corner case involving build order (manifests installing dependencies too late), and I've opened a bug for that at https://github.com/microsoft/vcpkg/issues/20830 . I think this issue is good to close. Thanks for the discussion and ideas.
